archetype-common/src/main/java/org/apache/maven/archetype/generator/DefaultFilesetArchetypeGenerator.java

```diff
@@ -136,7 +136,8 @@ public void generateArchetype( ArchetypeGenerationRequest request, File archetyp
 
             String packageName = request.getPackage();
             String artifactId = request.getArtifactId();
-            File outputDirectoryFile = new File( request.getOutputDirectory(), artifactId );
+            File outputDirectoryFile = new File( request.getOutputDirectory(),
```

Comment:
I think this makes more sense:

    File outputDirectoryFile = request.isCreateBaseDir() ? new File( request.getOutputDirectory(), artifactId): request.getOutputDirectory();

Reply:
`request.getOutputDirectory() ` return a String 

If i use your code , needs a bit change

```java
File outputDirectoryFile = request.isCreateBaseDir() ? new File( request.getOutputDirectory(), artifactId):  new File( request.getOutputDirectory() ) ;
```